lib/signature.js

```diff
@@ -0,0 +1,163 @@
+var crypto = require('crypto');
+var querystring = require('querystring');
+var sortKeys = require('sort-object');
+var Buffer = require('safe-buffer').Buffer;
+var scmp = require('scmp');
+var extend = Object.assign ? Object.assign : require('util')._extend;
+
+var MESSAGEBIRD_REQUEST_TIMESTAMP = 'messagebird-request-timestamp';
+var MESSAGEBIRD_REQUEST_SIGNATURE = 'messagebird-signature';
+var MESSAGEBIRD_REQUEST_TIMEOUT = 100;
+var MESSAGEBIRD_REQUEST_HASH = 'sha256';
+
+/**
+ * Returns request is expired or not.
+ *
+ * @param {Object} req
+ * @return {Boolean}
+ */
+function isRecent(req) {
+  var timestamp = req.headers[MESSAGEBIRD_REQUEST_TIMESTAMP];
+  var currentTime = Math.floor(new Date().getTime() / 1000);
+
+  if (!timestamp) {
+    throw new Error('The "MessageBird-Request-Timestamp" header is missing.');
+  }
+
+  return (currentTime - timestamp) < MESSAGEBIRD_REQUEST_TIMEOUT;
```

Comment:
This will probably work because JavaScript but should we check if the `timestamp` header is a valid number first (and cast it to a number) before doing this?

Reply:
👍